### PR TITLE
Odyssey Stats: Enable email sections and pages

### PR DESCRIPTION
**Pull request by kangzj**
Related to #87399

## Proposed Changes

* Enabled Email Stats card on Traffic page
* Enable Email Stats card on Subscriber page
* Enabled email list page, which lists recent emails
* Enabled email detail page, including Clicks and Opens
* Refactored support checks to a selector
* **Enable Email Stats for all site types on Calypso Stats, including Jetpack sites**

More information about Odyssey Stats testing instructions: PCYsg-Pp7-p2

## Testing Instructions

* Checkout branch `update/enable-email-stats-for-odyssey` and build Jetpack plugin - https://github.com/Automattic/jetpack/pull/35703
* Spin this change up on the local Jetpack environment: `cd apps/odyssey-stats && STATS_PACKAGE_PATH=/path-to-jetpack/projects/packages/stats-admin yarn dev`.
* Open Traffic page
* Ensure Email section works fine
* Click View details
* Ensure the Email list works okay
* Click an individual Email
* Ensure you are taken to Email Opens
* Ensure Email Opens page works okay

**!!Please also test with Jetpack trunk (older Jetpack), ensure the Email sections are still hidden, and it generally looks okay!!**

**!!Please test with Calypso Live Branch, and ensure Calypso Stats still works okay!!**

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

cc @Automattic/apex @Automattic/explorers @Automattic/gold @Automattic/zap @Automattic/red @lupus2k

### CODE REVIEW

File: client/my-sites/stats/pages/subscribers/index.tsx [hunk 1]
```diff
@@ -44,22 +39,16 @@ const StatsSubscribersPage = ( { period }: StatsSubscribersPageProps ) => {
...
+	const { supportsEmailStats, supportsSubscriberChart } = useSelector( ( state ) =>
+		getEnvStatsFeatureSupportChecks( state, siteId )
+	);
 	const today = new Date().toISOString().slice( 0, 10 );
 
 	const statsModuleListClass = classNames(
 		'stats__module-list stats__module--unified',
 		{
-			'is-odyssey-stats': isOdysseyStats,
+			'supports-email-stats': ! supportsEmailStats,
```
Comment (dognose24) [inline client/my-sites/stats/pages/subscribers/index.tsx]:
```suggestion
			'is-email-stats-unavailable': ! supportsEmailStats,
```

File: client/my-sites/stats/grid-layout.scss [hunk 2]
```diff
@@ -55,7 +55,7 @@ $grid-vertical-gutters: 32px;
 		display: none;
 	}
 
-	&.is-odyssey-stats {
+	&.supports-email-stats {
```
Comment (dognose24) [inline client/my-sites/stats/grid-layout.scss]:
```suggestion
	&.is-email-stats-unavailable {
```

File: client/my-sites/stats/highlights-section/index.tsx [hunk 3]
```diff
@@ -86,6 +86,7 @@ export default function HighlightsSection( {
 	);
 
 	// Highlights settings for Odyssey are not supported until stats-admin@0.9.0-alpha.
+	// TODO: refactor the checking to `client/state/sites/selectors/get-stats-feature-supports.ts`.
```
Comment (dognose24) [inline client/my-sites/stats/highlights-section/index.tsx]: 👍🏼

**Review by dognose24 — APPROVED** [review]
The change works as described on Calypso and Odyssey Stats and is compatible with old Jetpack versions. 👍🏼 I just left suggestions for naming, which are not blockers.

File: client/my-sites/stats/stats-email-detail/index.jsx [hunk 4]
```diff
@@ -342,7 +348,12 @@ const connectComponent = connect(
 	( state, ownProps ) => {
 		const { postId, statType, isValidStartDate } = ownProps;
 		const siteId = getSelectedSiteId( state );
-		const post = getSitePost( state, siteId, postId );
+		const postFallback = getPostStat( state, siteId, postId, 'post' );
```
Comment (dognose24) [inline client/my-sites/stats/stats-email-detail/index.jsx:351]: 👍🏼

File: client/my-sites/stats/post-detail-highlights-section/index.tsx [hunk 5]
```diff
@@ -58,16 +57,18 @@ export default function PostDetailHighlightsSection( {
...
-	const isAtomic = useSelector( ( state ) => siteId && isSiteWpcomAtomic( state, siteId ) );
-	const isWPcomSite = ! isJetpack || isAtomic;
+	const { supportsEmailStats } = useSelector( ( state ) =>
+		getEnvStatsFeatureSupportChecks( state, siteId )
+	);
 
 	// postId > 0: Show the tabs for posts except for the Home Page (postId = 0).
 	// isWPcomSite: The Newsletter Stats is only covering `WPCOM sites` for now.
 	// TODO: remove the (post?.date && new Date(post?.date) >= new Date("2023-05-30")) check when the Newsletter Stats data is backfilled.
 	const isEmailTabsAvailable =
```
Comment (kangzj) [inline client/my-sites/stats/post-detail-highlights-section/index.tsx:66]: Removed `isWPcomSite` checks here to make it available for all site types, including all Jetpack sites.

File: client/my-sites/stats/post-detail-highlights-section/index.tsx [hunk 6]
```diff
@@ -58,16 +57,18 @@ export default function PostDetailHighlightsSection( {
...
-
-	const isJetpack = useSelector( ( state ) => siteId && isJetpackSite( state, siteId ) );
-	const isAtomic = useSelector( ( state ) => siteId && isSiteWpcomAtomic( state, siteId ) );
-	const isWPcomSite = ! isJetpack || isAtomic;
+	const { supportsEmailStats } = useSelector( ( state ) =>
+		getEnvStatsFeatureSupportChecks( state, siteId )
+	);
 
 	// postId > 0: Show the tabs for posts except for the Home Page (postId = 0).
 	// isWPcomSite: The Newsletter Stats is only covering `WPCOM sites` for now.
```
Comment (kangzj) [inline client/my-sites/stats/post-detail-highlights-section/index.tsx]:
```suggestion
```

**Comment by kangzj** [comment]
I'll ship on the coming Monday if no further feedback received. Thanks all!

**Comment by kangzj** [comment]
Hi @crisbusquets, do you think we should hide the Emails Stats for site that don't have newsletter switched on?

File: client/state/sites/selectors/get-env-stats-feature-supports.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+const version_greater_than_or_equal = (
+	version: string | null,
+	compareVersion: string,
+	isOdysseyStats = false
+) => {
+	return !! ( ! isOdysseyStats || ( version && version_compare( version, compareVersion, '>=' ) ) );
+};
+
+export default function getEnvStatsFeatureSupportChecks( state: object, siteId: number | null ) {
```
Comment (a8ck3n) [inline client/state/sites/selectors/get-env-stats-feature-supports.ts:13]: Very nice addition here!

**Review by a8ck3n — APPROVED** [review]
Tests well for me!

**Comment by crisbusquets** [comment]
> Hi @crisbusquets, do you think we should hide the Emails Stats for site that don't have newsletter switched on?

Hi @jsnmoon! Sorry, I missed this ping. What did you decide?

**Comment by kangzj** [comment]
Hi @crisbusquets, all good. Email Stats is always on right now even if Newsletter is not turned on for Jetpack sites (probably for wpcom sites too - I need to confirm). What do you think?

**Comment by crisbusquets** [comment]
> Email Stats is always on right now even if Newsletter is not turned on for Jetpack sites (probably for wpcom sites too - I need to confirm). What do you think?

I like that it's always on, but we need to use the empty state to let users know about newsletters. The current state is this one (at least on a JN site):
image

image

Click `your emails` opens this: https://wordpress.com/support/stats/understand-your-sites-traffic/#emails

Can we implement a launchpad like the one we show when there are no subscribers (this is WPcom)? But with different tasks (not sure which ones yet):
image

@ederrengifo what do you think?

cc @simison for awareness

**Comment by kangzj** [comment]
Thanks @crisbusquets 👍 

> Can we implement a launchpad like the one we show when there are no subscribers (this is WPcom)? But with different tasks (not sure which ones yet):

We have https://github.com/Automattic/wp-calypso/pull/87657 for this! about to merge anytime now 😄

> I like that it's always on, but we need to use the empty state to let users know about newsletters. The current state is this one (at least on a JN site):

Good point. We have a https://github.com/Automattic/jetpack-roadmap/issues/870 to consider empty states for all components, but we could do it piece by piece if y'all have suggestions! @crisbusquets  @ederrengifo

**Pull request merged**